osu.Game/Screens/Edit/GameplayTest/EditorPlayer.cs

```diff
@@ -56,6 +62,32 @@ protected override void LoadComplete()
             });
         }
 
+        private void markPreviousObjectsHit()
+        {
+            foreach (var hitObject in enumerateHitObjects(DrawableRuleset.Objects, editorState.Time))
+            {
+                var judgement = hitObject.Judgement;
+
+                HealthProcessor.ApplyResult(new JudgementResult(hitObject, judgement) { Type = judgement.MaxResult });
+                ScoreProcessor.ApplyResult(new JudgementResult(hitObject, judgement) { Type = judgement.MaxResult });
+            }
```

Comment:
If the problem with the replay frame-based approach is flawed because it relied on using hitobjects and hitobject-based cutoffs, then how is _this_ fine? Why is there no attempt to compensate for preempt or transforms or whatever?

That explanation isn't making much sense to me in light of this.

Reply:
The hookup flow with `ScoreProcessor` via `Player` means that applying result here will have the results end up on the DHOs, seemingly. It's a long chain of events that bugs me every time i have to traverse it.